samples/bluetooth/ble_hids_keyboard/src/main.c

```diff
@@ -0,0 +1,1023 @@
+#include <bluetooth/peer_manager/peer_manager_handler.h>
+#include <event_scheduler.h>
+#include <zephyr/sys/ring_buffer.h>
+#include <zephyr/toolchain.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/logging/log_ctrl.h>
+#include <board-config.h>
+
+LOG_MODULE_REGISTER(app, CONFIG_BLE_HIDS_KEYBOARD_SAMPLE_LOG_LEVEL);
+
+#define BASE_USB_HID_SPEC_VERSION 0x0101
+
+/* Control key codes - required 8 of them */
+#define INPUT_REPORT_KEYS_CTRL_CODE_MIN 224
+/* Control key codes - required 8 of them */
+#define INPUT_REPORT_KEYS_CTRL_CODE_MAX 231
+
+/* Index of Input Report. */
+#define INPUT_REPORT_KEYS_INDEX 0
+/* Id of reference to Keyboard Input Report. */
+#define INPUT_REP_REF_ID 0
+
+/* Index of Output Report. */
+#define OUTPUT_REPORT_INDEX 0
+/* Id of reference to Keyboard Output Report. */
+#define OUTPUT_REP_REF_ID 0
+/* Index of Feature Report. */
+#define FEATURE_REPORT_INDEX 0
+/* Id of reference to Keyboard Feature Report. */
+#define FEATURE_REP_REF_ID 0
+/* CAPS LOCK bit in Output Report (based on 'LED Page (0x08)' of the
+ * Universal Serial Bus HID Usage Tables).
+ */
+#define OUTPUT_REPORT_BIT_MASK_CAPS_LOCK 0x02
+
+/* Perform bonding. */
+#define SEC_PARAM_BOND 1
+/* Man In The Middle protection not required. */
+#define SEC_PARAM_MITM 0
+/* LE Secure Connections enabled. */
+#define SEC_PARAM_LESC 1
+/* Keypress notifications not enabled. */
+#define SEC_PARAM_KEYPRESS 1
+/* No I/O capabilities. */
+#define SEC_PARAM_IO_CAPABILITIES BLE_GAP_IO_CAPS_DISPLAY_YESNO
+/* Out Of Band data not available. */
+#define SEC_PARAM_OOB 0
+/* Minimum encryption key size. */
+#define SEC_PARAM_MIN_KEY_SIZE 7
+/* Maximum encryption key size. */
+#define SEC_PARAM_MAX_KEY_SIZE 16
+
+/** Battery Level sensor simulator state. */
+static struct sensorsim_state battery_sim_state;
+/** Battery timer. */
+static struct bm_timer battery_timer;
+
+/* HID service instance */
+BLE_HIDS_DEF(ble_hids);
+/* BLE Advertising library instance */
+BLE_ADV_DEF(ble_adv);
+/* BLE QWR instance */
+BLE_QWR_DEF(ble_qwr);
+/* BLE BAS instance */
+BLE_BAS_DEF(ble_bas);
+
+/* BLE Connection handle */
+static uint16_t conn_handle = BLE_CONN_HANDLE_INVALID;
+/* Peer ID */
+static pm_peer_id_t peer_id;
+/* Flag for ongoing authentication request */
+static bool auth_key_request;
+/* State of caps lock */
+static bool caps_on;
+
+/* Forward declaration */
+static void identities_set(pm_peer_id_list_skip_t skip);
+
+/* FIFO for keeping track of keystrokes that can not be sent immediately. */
+RING_BUF_DECLARE(report_fifo, CONFIG_BLE_HIDS_REPORT_FIFO_SIZE *
+		 (sizeof(struct ble_hids_input_report) + CONFIG_BLE_HIDS_INPUT_REPORT_MAX_LEN));
+
+uint32_t report_fifo_put(struct ble_hids_input_report *report)
+{
+	uint32_t written;
+	uint32_t space;
+
+	space = ring_buf_space_get(&report_fifo);
+
+	if (space < (sizeof(struct ble_hids_input_report) + CONFIG_BLE_HIDS_INPUT_REPORT_MAX_LEN)) {
+		LOG_ERR("Could not put input report in buffer");
```

Comment:
This failure is different than the one below, so I would argue that the message should be different as well, more like "Not enough space for report ...."